src/actions/actions.ts

```diff
@@ -3537,17 +3537,52 @@ class ToggleCaseOperator extends BaseOperator {
 
   public async run(vimState: VimState, start: Position, end: Position): Promise<VimState> {
     const range = new vscode.Range(start, new Position(end.line, end.character + 1));
-    const char = TextEditor.getText(range);
-
-    // Try lower-case
-    let toggled = char.toLocaleLowerCase();
-    if (toggled === char) {
-      // Try upper-case
-      toggled = char.toLocaleUpperCase();
+    const text = TextEditor.getText(range);
+
+    let newText = "";
+    for (var i = 0; i < text.length; i++) {
+      var char = text[i];
+      // Try lower-case
+      let toggled = char.toLocaleLowerCase();
+      if (toggled === char) {
+        // Try upper-case
+        toggled = char.toLocaleUpperCase();
+      }
+      newText += toggled;
     }
+    await TextEditor.replace(range, newText);
+
+    const cursorPosition = start.isBefore(end) ? start : end;
+    vimState.cursorPosition = cursorPosition;
+    vimState.cursorStartPosition = cursorPosition;
+    vimState.currentMode = ModeName.Normal;
+
+    return vimState;
+  }
+}
 
-    if (toggled !== char) {
-      await TextEditor.replace(range, toggled);
+@RegisterAction
+class ToggleCaseVisualBlockOperator extends BaseOperator {
+  public keys = ["~"];
+  public modes = [ModeName.VisualBlock];
+
+  public async run(vimState: VimState, start: Position, end: Position): Promise<VimState> {
+    for (const { start, end } of Position.IterateLine(vimState)) {
+      const range = new vscode.Range(start, end);
+      const text = TextEditor.getText(range);
+
+      let newText = "";
+      for (var i = 0; i < text.length; i++) {
```

Comment:
Haha, yeah, you know it had to happen. Maybe in a utilities file somewhere? Or if you can't find/dont want to make one, then a static method on a class can work.